__init__.py

```
@@ -32,25 +33,54 @@ def initialize(self):
+        self.camera_type = self.detect_camera_type()
 
-    def open_camera(self) -> VideoStream:
-        if self._camera is None:
+    @staticmethod
+    def detect_camera_type() -> str:
+        """Detect if running on Raspberry Pi with libcamera or desktop."""
+        try:
+            # only works in rpi
+            import libcamera
+            return "libcamera"
+        except:
```

Ruff

Do not use bare `except`

(E722)